js/src/logger.ts

```diff
@@ -1642,6 +1655,87 @@ function validateAndSanitizeExperimentLogFullArgs(
   return event;
 }
 
+class ObjectFetcher<RecordType> {
+  private excludeFields: Set<string> = new Set(["_object_delete"]);
+  private _fetchedData: any[] | undefined = undefined;
```

Comment:
Why isn't the type `RecordType[] | undefined`?

Reply:
it has to be post-processed in `fetch()` before it's `RecordType`